src/components/LeftSidebar/SidebarContent.astro

```diff
@@ -39,10 +44,20 @@ const lang = getLanguageFromURL(Astro.canonicalURL.pathname);
 					</li>
 				))}
 			</ul>
-		</div>
+		</details>
 	</li>
 ))}
 
+<script>
+	const isMobileView = window.matchMedia('(max-width: 50em)').matches;
+
+	if (isMobileView) {
+		document.querySelectorAll('aside details').forEach((e) => {
+			e.removeAttribute('open');			
+		});
+	}
+</script>
+
```

Comment:
What do you think about moving this logic to `Header/SidebarToggle.tsx`? We can collapse the details each time the menu gets opened on mobile. It’s a tiny bit of redundancy, but it gets us around the weird doesn’t work on viewport resize issue, plus that component is loaded using `client:idle`.

Here’s the `useEffect` hook from that component updated with your logic (I’ve also scope the CSS selector slightly differently):

```js
	useEffect(() => {
		const body = document.getElementsByTagName('body')[0];
		if (sidebarShown) {
			body.classList.add('mobile-sidebar-toggle');
			document.querySelectorAll('.nav-group details').forEach((e) => {
				e.removeAttribute('open');
			});
		} else {
			body.classList.remove('mobile-sidebar-toggle');
		}
	}, [sidebarShown]);
```

Reply:
That's a neat solution, glad we have your smart brain 🙌. What about adding the 'open' attribute back when we close the sidebar toggle? This way if I go to the mobile view, open the toggle, close it and go back to the desktop view I have all the sections open again. This will be rarely noticed but I just felt like "why not?".

Reply:
Sure, don’t see why not!